rotkehlchen/tests/db/test_db_upgrades.py

```diff
@@ -3581,10 +3596,25 @@ def test_upgrade_db_48_to_49(user_data_dir, messages_aggregator):
         assert 'TEXT_NOT NULL' not in schema_info[0], 'TEXT_NOT NULL bug should be fixed'
         assert 'to_amount TEXT NOT NULL' in schema_info[0], 'Should have correct TEXT NOT NULL syntax'  # noqa: E501
 
-        # Check that data was preserved
-        new_swaps_data = cursor.execute(
-            'SELECT tx_id, from_asset, from_amount, to_asset, to_amount FROM zksynclite_swaps ORDER BY tx_id',  # noqa: E501
-        ).fetchall()
-        assert new_swaps_data == swaps_data, 'Swap data should be preserved after upgrade'
+        # Test solana assets migration to CAIPS format
+        # verify old format assets are gone
+        assert cursor.execute('SELECT COUNT(*) FROM assets WHERE identifier IN ("$NAP", "ACS", "AI16Z", "BONK") ORDER BY identifier').fetchone()[0] == 0  # noqa: E501
+        # verify duplicates were deleted
+        assert cursor.execute('SELECT COUNT(*) FROM assets WHERE identifier IN ("TRISIG", "HODLSOL")').fetchone()[0] == 0  # noqa: E501
+        # Check that asset references were updated in other tables
+        assert cursor.execute('SELECT currency, amount, usd_value FROM timed_balances WHERE currency LIKE "solana%"').fetchall() == [  # noqa: E501
+            ('solana/token:4G86CMxGsMdLETrYnavMFKPhQzKTvDBYGMRAdVtr72nu', '100.5', '15000.0'),
```

Comment:
perhaps it;s best to keep the expected balances tuples list as a separate variable. And then have a before and after mapping for the identifiers of the tokens instea.d Same for other tables. Do you understand what I mean?